src/main/java/xbot/common/trajectory/LowResField.java

```diff
@@ -24,13 +24,306 @@ public LowResField() {
+                                pointClosestToInside,
+                                currentTarget.getRotation2d(),
+                                10
+                        );
+                        swervePointStack.push(swervePointNearInside);
+                        currentTarget = swervePointNearInside;
+                    }
+                    else {
+                        log.info("Handling non-parallel crossing; typical case..");
+                        // General, most straightforward case. Get the nearest corner to the average intersection,
+                        // and prepare for another loop.
+                        var nearestCorner = o.getClosestCornerToPoint(averageIntersection);
+                        XbotSwervePoint cornerPoint = XbotSwervePoint.createXbotSwervePoint(
+                                nearestCorner,
+                                currentTarget.getRotation2d(),
+                                10
+                        );
+                        log.info("Chosen corner:" + nearestCorner.toString());
+                        swervePointStack.push(cornerPoint);
+                        currentTarget = cornerPoint;
+                    }
+
+                    break;
+                }
+            }
+
+            // We either just collided with something, or checked all obstacles and found nothing.
+            // If we didn't collide with anything, we need to check for terminating or updating conditions.
+            if (!collision) {
+                if (swervePointStack.size() > 0) {
+                    // 4b) If no collision, and the stack isn't empty, fully pop the stack into the list and set
+                    //         the current target to the ultimate target, and the current position to the last point in the list.
+                    //         Loop back to step (3).
+                    while (swervePointStack.size() > 0) {
+                        path.add(swervePointStack.pop());
+                    }
+                    log.info("Popped stack into path.");
+                    log.info("Setting target to ultimate target");
+
+                    currentTarget = ultimateTarget;
+                    currentSource = path.get(path.size() - 1).getTranslation2d();
+
+                    log.info("Set source to last point in path: " + currentSource.toString());
+                } else {
+                    log.info("No more collisions found; route clear. Adding terminating point and finishing.");
+                    // 4a) If no collision, AND the current target is the ultimate target, we are done.
+                    //         Add the current target to the list, and return.
+                    path.add(currentTarget);
+
+                    if (specialFinalPoint != null) {
+                        path.add(specialFinalPoint);
+                    }
+
+                    completed = true;
+                }
+            }
+
+        }
+        return path;
```

Comment:
this is a pretty enormous function (not just from the comments), what do you think about breaking it down into a few smaller pieces so the outer function reads a little easier?

I also wonder if those inner parts wouldn't be easier to unit test as well?